docs/git/index.md

```diff
@@ -0,0 +1,295 @@
+
+To avoid this, we use something called a _fork_ and something called a _pull request_.
+
+Remember when we cloned the repo, we pasted `https://github.com/CodeYourFuture/cyf-demo-repo` in as the place to clone from? Let's look at that:
+
+TODO: Insert image annotating bits of the URL
+
+This is saying "On GitHub, the user `CodeYourFuture` has a repository called `cyf-demo-repo`, I want that".
+
+GitHub lets you host your own copy of the repository on GitHub too! If your username is `EagerLearner`, can you guess what URL your repository would be at?
+
+That's right, https://github.com/EagerLearner/cyf-demo-repo!
+
+This is called a _fork_. It's a copy of the repository, where you're allowed to make changes. So when GitHub Desktop just asked us "Do you want to fork this repository?", what it's really saying is "You're not allowed to make changes to CodeYourFuture's repository, would you like to make your own copy on GitHub where you _are_ allowed to make changes, and put your changes there?"
+
+That sounds like exactly what we want to do, so we'll click the "Fork This Repository" button.
+
+Then GitHub asks us whether we want to fork "To contribute to the parent project" (i.e. because we want to work with CodeYourFuture) or "For my own purposes" (i.e. because we want to do our own thing apart from CodeYourFuture). We want to work with CodeYourFuture, so we'll select "To contribute to the parent project" and press Continue.
+
+Now if we press "Push origin", it will copy our changes to our fork on GitHub.
+
+TODO: Insert graphic showing four repositories: Teacher's computer, CodeYourFuture, EagerLearner Fork, EagerLearner's computer.
+
+If you forget this, it's in the cheatsheet (TODO: Link to cheatsheet)! Check out the "I want to send my code to volunteers (Pushing)" section.
+
+## Making a Pull Request
+
+Now that we've pushed our homework to our fork, we need to tell CodeYourFuture about it, so that the volunteers know to look at it!
+
+We do this with something called a _pull request_. This is a slightly weird name, in the context of homework.
+
+Normally, when people push changes to a fork on GitHub, they're doing so because they want the person who owns the repository to look at the changes, and pull the changes into their repository. For example, this webpage we're reading the syllabus on right now is hosted on GitHub, and if someone spots a typo, they can fix it, push it to their fork, and request for CodeYourFuture to pull their change into CodeYourFuture's version (hence the name a pull request - "requesting to pull in changes").
+
+We call pulling someone's change into a repository "merging" the change (some people use the phrase "merge request" instead of "pull request").
```

Comment:
Not sure this sentence is necessary? I feel like the paragraph above explains the naming well enough, and the specifics of merging aren't necessary.

Reply:
I added the "merging" because it feels like "merge" is a verb-of-art they're going to see soon enough, and it may be a bit confusing if we've explained "They will pull in your changes" as the terminology, to not get the word "merge"... Maybe keeping the sentence, but dropping the parenthesised bit?

Reply:
Thanks, I agree with removing the parenthesised bit. Perhaps merging (!) with the next paragraph too? Feels like that would flow better?